src/core/account.ts

```diff
@@ -146,6 +145,42 @@ export class Account {
     return new Account({ privateKey, address, legacy: args?.legacy });
   }
 
+  /**
+   * Instantiates an account given a private key.
+   *
+   * This is used a local calculation and therefore is used to instantiate an `Account`
+   * that has not had its authentication key rotated.
+   *
+   * @param privateKey PrivateKey - private key of the account
+   * @param args.legacy optional. If set to true, the keypair generated is a Legacy keypair. Defaults
+   * to generating a Unified keypair
+   *
+   * @returns Account
+   */
+  static fromPrivateKey(args: { privateKey: PrivateKey; legacy?: boolean }): Account {
+    const { privateKey, legacy } = args;
+
+    let publicKey;
+    if (privateKey instanceof Secp256k1PrivateKey) {
+      // Secp256k1 single sender
+      publicKey = new AnyPublicKey(privateKey.publicKey());
+    } else if (privateKey instanceof Ed25519PrivateKey) {
+      // legacy Ed25519
+      if (legacy) {
+        publicKey = privateKey.publicKey();
+      } else {
+        // Ed25519 single sender
+        publicKey = new AnyPublicKey(privateKey.publicKey());
+      }
+    } else {
+      throw new Error(`Unsupported private key ${privateKey}`);
+    }
+
+    const authKey = AuthenticationKey.fromPublicKey({ publicKey });
+    const address = new AccountAddress({ data: authKey.toUint8Array() });
+    return new Account({ privateKey, address, legacy });
+  }
```

Comment:
Is there a reason why not to make this and `PrivateKeyAndAddress` the same function, and just have an optional address?  This is pretty much the constructor just that it autogenerates the account address.

Maybe we can make all of these (except from mnemonic) into the constructor and remove any choices needed to be made between these 3 very similar paths?

It would be good to explore how to deal with auto lookups the best way (I'm aware of the function on `Aptos`) and reduce down the number of ways to create an Account as much as possible

Reply:
tbh, I dont see any reason to have one function for both options. In general I am not a fan of optional arguments and to try and squish as much arguments into one function. It becomes messy really fast and explain what is going on is just confusing. Also, those 2 functions are pretty different in their usage, `fromPrivateAndAddress` cares for key rotation while `fromPrivateKey` doesnt.